app/src/main/java/org/kiwix/kiwixmobile/zim_manager/fileselect_view/effects/DeleteFiles.kt

```diff
@@ -40,18 +40,24 @@ data class DeleteFiles(private val booksOnDiskListItem: List<BookOnDisk>) :
 
   override fun invokeWith(activity: AppCompatActivity) {
     (activity as ZimManageActivity).cachedComponent.inject(this)
-    booksOnDiskListItem.forEach {
-      dialogShower.show(DeleteZim(it), {
-        if (deleteSpecificZimFile(it)) {
-          if (it.file.canonicalPath == zimReaderContainer.zimCanonicalPath) {
-            zimReaderContainer.setZimFile(null)
-          }
-          activity.toast(R.string.delete_specific_zim_toast)
-        } else {
-          activity.toast(R.string.delete_zim_failed)
-        }
-      })
-    }
+
+    val name = booksOnDiskListItems.joinToString(separator = "\n") { it.book.title }
+
+    dialogShower.show(DeleteZims(name), {
+      val booksDeleted = booksOnDiskListItems.fold(true) { acc, book ->
+        acc &&
```

Comment:
we want this to be a little more concise, a thing to look for is if you a returning a boolean value based on an if statement, it will always be more clear to just write the statement
eg
```
if (delete())
return true
else 
return false
```
is conciser as
```
return delete()
```

Your problem here is you want to do this side effect sort of thing, for that scenario I like to use `also`

Reply:
On it. Sorry, I missed that part the last comment.